app/models/setting/provisioning.rb

```diff
@@ -20,9 +20,20 @@ def self.load_defaults
         self.set('libvirt_default_console_address', N_("The IP address that should be used for the console listen address when provisioning new virtual machines via Libvirt"), "0.0.0.0"),
         self.set('update_ip_from_built_request', N_("Foreman will update the host IP with the IP that made the built request"), false),
         self.set('use_shortname_for_vms', N_("Foreman will use the short hostname instead of the FQDN for creating new virtual machines"), false),
+        self.set('websockets_encrypt', N_("VNC/SPICE websocket proxy console access encryption (websockets_ssl_key/cert setting required)"), SETTINGS[:require_ssl]),
+        self.set('websockets_ssl_key', N_("Private key that Foreman will use to encrypt websockets "), nil),
+        self.set('websockets_ssl_cert', N_("Certificate that Foreman will use to encrypt websockets "), nil)
       ].each { |s| self.create! s.update(:category => "Setting::Provisioning")}
     end
 
     true
   end
+  
+  def validate_websockets_encrypt(record)
+    if Setting["websockets_ssl_key"].empty? || Setting["websockets_ssl_cert"].empty?
+      # we cannot fail due to compatibility so we force set it here
+      Rails.logger.warn("Either websockets_ssl_key or websockets_ssl_cert is missing, turning off websockets_encrypt option.")
```

Comment:
I was thinking migration, but I should be able to workaround it I guess. During migration, if key/cert is missing and encrypt is on, I can turn it off apparently.